lib/rules/no-extra-parens.js

```diff
@@ -169,6 +169,28 @@ module.exports = {
             return ruleApplies(node) && isParenthesisedTwice(node);
         }
 
+        /**
+         * Determines if a node that is expected to be parenthesised is surrounded by
+         * (potentially) invalid extra parentheses with considering precedence level of the node.
+         * If the preference level of the node is higher or equal to precedence lower limit, it checks
+         * whether the node is surrounded by parentheses twice or not.
```

Comment:
This sounds more like `&&` than `||` :-)

Reply:
@mdjermanovic 
yes :) you are right. I will change the comments. Thanks 👍